### ISSUE TITLE
`reportUnusedVariable` false positives when used with an `Unknown` list

### ISSUE THREAD

**Issue by Avasam** [microsoft/pyright#3545]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
In the example below, `read` is unknown. So `image` is unknown as well. Pyright doesn't see that I do use the variable `x` to slice or access the `image`. Note this doesn't happen at the top level, it has to be inside of a function.

**To Reproduce**
See minimal repro below

**Expected behavior**
No errors

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
import cv2


def __example(capture_device: cv2.VideoCapture):
    _, image = capture_device.read()
    x = 1
    image[x]
```

![image](https://user-images.githubusercontent.com/1350584/172193507-702b5512-f904-49ca-95f0-a25a15c95958.png)

> C:\Users\Avasam\Documents\Git\Auto-Split\src\region_capture.py
  C:\Users\Avasam\Documents\Git\Auto-Split\src\region_capture.py:139:5 - error: Variable "y" is not accessed (reportUnusedVariable)
  C:\Users\Avasam\Documents\Git\Auto-Split\src\region_capture.py:140:5 - error: Variable "x" is not accessed (reportUnusedVariable)

~~I don't think it's specific to cv2, but~~ I'm using the types bundled with `microsoft/python-type-stubs`

**VS Code extension or command-line**
pyright@1.1.252

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release. It stems from a bug in the type stubs where `VideoCapture` is declared in a circular fashion. 

```
VideoCapture = _mod_cv2.VideoCapture
```

Since `_mod_cv2` refers back to the same type stub, this is effectively the same as

```
VideoCapture = VideoCapture
```

Pyright detects this case and reports an error, but it didn't properly handle the resulting type, and this propagated up to eventually cause the problem you observed.

**Comment by erictraut** [comment]
I've also filed [this bug](https://github.com/microsoft/python-type-stubs/issues/213) to report the circular type aliases in the stub.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.253, which I just published. It will also be included in the next release of pylance.
